src/test/java/com/redislabs/redisgraph/RedisGraphAPITest.java

```diff
@@ -12,247 +13,77 @@ public RedisGraphAPITest() {
+    	
+    	Assert.assertEquals("1", result.getStatistics().getStringValue(Label.NODES_CREATED));
+    	Assert.assertEquals("2", result.getStatistics().getStringValue(Label.PROPERTIES_SET));
+    	Assert.assertNotNull(result.getStatistics().getStringValue(Label.QUERY_INTERNAL_EXECUTION_TIME)); 
     }
 
-    @org.testng.annotations.Test
+    @Test
     public void testConnectNodes() throws Exception {
         // Create both source and destination nodes
-        RedisNode src = api.createNode("name", "roi", "age", 32);
-        RedisNode dest = api.createNode("name", "amit", "age", 30);
-
-        // Connect source and destination nodes.
-        RedisEdge edge = api.connectNodes(src, "knows", dest, "strength", "3", "from", "high-school");
-
-        // Validate edge
-        Assert.assertNotNull(edge.getId());
-        Assert.assertEquals(edge.getRelation(), "knows");
-        Assert.assertSame(edge.getAttributes().size(), 2);
-        Assert.assertSame(edge.getAttributes().get("strength"), "3");
-        Assert.assertSame(edge.getAttributes().get("from"), "high-school");
-        Assert.assertSame(edge.getSrc(), src);
-        Assert.assertSame(edge.getDest(), dest);
+    	ResultSet createResult1 = api.query("CREATE (:person{name:'roi',age:32})");
+    	ResultSet createResult2 = api.query("CREATE (:person{name:'amit',age:30})");
+    	
+    	// Connect source and destination nodes.
+    	ResultSet matchResult = api.query("MATCH (a:person), (b:person) WHERE (a.name = 'roi' AND b.name='amit')  CREATE (a)-[knows]->(a)");
+    	
+    	Assert.assertFalse(matchResult.hasNext());
+    	Assert.assertNull(matchResult.getStatistics().getStringValue(Label.NODES_CREATED));
+    	Assert.assertNull(matchResult.getStatistics().getStringValue(Label.PROPERTIES_SET));
+    	Assert.assertEquals("1", matchResult.getStatistics().getStringValue(Label.RELATIONSHIPS_CREATED));
+    	Assert.assertNull("0", matchResult.getStatistics().getStringValue(Label.RELATIONSHIPS_DELETED));
+    	Assert.assertNotNull(matchResult.getStatistics().getStringValue(Label.QUERY_INTERNAL_EXECUTION_TIME)); 
     }
 
-    @org.testng.annotations.Test
+    @Test
     public void testQuery() throws Exception {
-        // Create both source and destination nodes
-        RedisNode src = api.createLabeledNode("qhuman","name", "roi", "age", 32);
-        RedisNode dest = api.createLabeledNode("qhuman","name", "amit", "age", 30);
-
-        // Connect source and destination nodes.
-        api.connectNodes(src, "knows", dest, "strength", "3", "from", "high-school");
+    	
+        // Create both source and destination nodes    	
+    	ResultSet create1Result = api.query("CREATE (:qhuman{name:'roi',age:32})");
+    	ResultSet create2Result = api.query("CREATE (:qhuman{name:'amit',age:30})");
+    	
+    	// Connect source and destination nodes.
+    	ResultSet connectResult= api.query("MATCH (a:qhuman), (b:qhuman) WHERE (a.name = 'roi' AND b.name='amit')  CREATE (a)-[knows]->(b)");
```

Comment:
what's a qhuman, can't we simply use human?

Reply:
I followed the original tests, and since we don't really have Graph drop implementation I want to use different keys